lib/system/database/definitions/mysql.rb

```diff
@@ -63,12 +63,6 @@
     SQL
   end
 
-  trigger 'end_user_plans' do
```

Comment:
This means that we must recreate the triggers. In SaaS for sure and probably in on-prem too.

We made it with a migration about 1 year ago in https://github.com/3scale/porta/pull/1047
That would work both for SaaS and for on-prem. But I do not know if the migration is blocking... which by then didn't matter because nobody was using that table anyway, but we cannot block the services table so it is not the same XD **if** it is blocking (I remark the IF because I do not know), we shouldn't make it this way.

We could also open [an issue for Ops](https://github.com/3scale/infrastructure/issues) to recreate triggers in SaaS themselves then :smile: You have an example in https://github.com/3scale/infrastructure/issues/759
But that would be only for SaaS and we should think what to do for on-prem (maybe they should execute `Rake::Task['db:triggers'].invoke`  but probably we should make a migration for them that is not executed in SaaS so it is much more efficient).

Reply:
Sorry, it is in the end_user_plans table, my bad, so we can just do the migration no matter if it is blocking or not because nobody uses that table :wink:

Reply:
We spoke about this more in-depth in private :+1:

Reply:
I agree that normally the table is not accessed but for security it is preferable to have this trigger deleted before this PR, so we in case something bad happens like some unlikely queries with invalid data, it won't just stop the request to finish.

Reply:
Here you go! #1738